driver/src/main/java/org/neo4j/driver/v1/Values.java

```diff
@@ -27,6 +27,7 @@
 import java.util.Map;
 
 import org.neo4j.driver.v1.exceptions.ClientException;
+import org.neo4j.driver.v1.internal.util.Extract;
```

Comment:
I really think we should enforce strict separation here - the API should never explicitly depend on the implementation. If we take that stance, it follows that we'll need to change `Values` somehow.

One way to do that is to make the `Values` utility into an interface, and allowing users to get instances of it from `Driver` or `GraphDatabase`. 

Another is that we could move all the simple value classes into the API as well - SimpleNode, SimpleMap, SimpleBlah all become part of the API.

Yet another is to make `Values` be an implementation utility - it's there, but it is not part of the API. That seems less appealing though, since it'd kill the portability of pretty much any app using the API. 

Maybe there are other options as well - in any case, one of the biggest problems with the current Core API is that it is circularly dependent on its implementation. We should enforce in the build that that is not the case for the new API.